src/Cli/ConfigGenerator.cs

```diff
@@ -918,7 +938,150 @@ private static bool TryUpdateConfiguredCacheValues(
             }
             catch (Exception ex)
             {
-                _logger.LogError("Failure in updating RuntimeConfig.Cache with exception message: {exceptionMessage}.", ex.Message);
+                _logger.LogError("Failed to update RuntimeConfig.Cache with exception message: {exceptionMessage}.", ex.Message);
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Attempts to update the Config parameters in the Host runtime settings based on the provided value.
+        /// Validates that any user-provided parameter value is valid and then returns true if the updated Host options
+        /// needs to be overwritten on the existing config parameters
+        /// </summary>
+        /// <param name="options">options.</param>
+        /// <param name="updatedHostOptions">updatedHostOptions.</param>
+        /// <returns>True if the value needs to be udpated in the runtime config, else false</returns>
+        private static bool TryUpdateConfiguredHostValues(
+            ConfigureOptions options,
+            ref HostOptions? updatedHostOptions)
+        {
+            object? updatedValue;
+            try
+            {
+                // Runtime.Host.Mode
+                updatedValue = options?.RuntimeHostMode;
+                if (updatedValue != null)
+                {
+                    updatedHostOptions = updatedHostOptions! with { Mode = (HostMode)updatedValue };
```

Comment:
is `updatedHostOptions` guaranteed to be not null here? If yes, why do we have the argument to the function defined to be nullable?